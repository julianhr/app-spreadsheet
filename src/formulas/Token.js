class Token {
  constructor(type, text, whitespace='') {
    this.type = type
    this.text = text
    this.value = null
    this.html = null
    this.whitespace = whitespace
  }

  _repr() {
    return Object
      .entries(this)
      .reduce((prev, curr) => {
        if (curr[1] !== null) {
          prev[curr[0]] = curr[1]
        }

        return prev
      }, {})
  }
}

export default Token
