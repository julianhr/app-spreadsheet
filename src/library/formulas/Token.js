class Token {
  constructor(type, text, whitespace='', value=null) {
    this.type = type
    this.text = text
    this.value = value
    this.whitespace = whitespace
  }

  _repr() {
    const { text, value, type, whitespace } = this
    return { text, value, type, whitespace }
  }
}

export default Token
