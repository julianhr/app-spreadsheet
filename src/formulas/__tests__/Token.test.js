import Token from '../Token'


describe('Token', () => {
  it('instantiates correctly', () => {
    const type = 'NUMBER'
    const text = '2'
    const whitespace = 1
    const category = 'entity'
    const index = 0

    const token = new Token(type, text, whitespace, category, index)
    expect(token).toMatchSnapshot()
  })
})
