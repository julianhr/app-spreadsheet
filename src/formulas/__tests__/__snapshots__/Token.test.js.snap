// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Token instantiates correctly 1`] = `
Token {
  "category": "entity",
  "html": null,
  "index": 0,
  "text": "2",
  "type": "NUMBER",
  "value": null,
  "whitespace": 1,
}
`;
